File: src/Apps/Logistics.OfficeApp/src/app/shared/models/load.ts
import { LoadStatus } from './loadStatus';

export interface Load {
  id?: string;
  refId?: number;
  name?: string;
  sourceAddress?: string;
  destinationAddress?: string;
  deliveryCost?: number;
  distance?: number;
  status?: LoadStatus;
  dispatchedDate?: string;
  pickUpDate?: string;
  deliveryDate?: string;
  assignedDispatcherId?: string;
  assignedDispatcherName?: string;
  assignedTruckId?: string;
  assignedDriverId?: string;
  assignedDriverName?: string;
}

File: src/Client/Logistics.Sdk/Models/Load.cs
using Logistics.Sdk.Enums;

namespace Logistics.Sdk.Models;

public record Load
{
    public string? Id { get; set; }
    public ulong RefId { get; set; } = 100_000;
    public string? Name { get; set; }
    public string? SourceAddress { get; set; }
    public string? DestinationAddress { get; set; }
    public double DeliveryCost { get; set; }
    public double Distance { get; set; }
    public DateTime DispatchedDate { get; set; }
    public DateTime? PickUpDate { get; set; }
    public DateTime? DeliveryDate { get; set; }
    public LoadStatus Status { get; set; }
    public string? AssignedDispatcherId { get; set; }
    public string? AssignedDispatcherName { get; set; }
    public string? AssignedDriverId { get; set; }
    public string? AssignedDriverName { get; set; }
    public string? AssignedTruckId { get; set; }
}
